cpp/src/strings/json/json_path.cu

```diff
@@ -97,80 +98,48 @@ class parser {
     return false;
   }
 
+  /**
+   * @brief Parse a quote-enclosed JSON string.
+   *
+   * @param[out] str The resulting string.
+   * @param can_be_empty Parameter indicating whether it is valid for the string
+   * to not be present.
+   * @param quote Character expected as the surrounding quotes.  A value of 0
+   * indicates allowing either single or double quotes (but not a mixture of both).
+   * @returns A result code indicating success, failure or other result.
+   */
   CUDA_HOST_DEVICE_CALLABLE parse_result parse_string(string_view& str,
```

Comment:
I'm not sure if it is too late to comment here: This C-style function call (returns error code while the output is a reference parameter) should be avoided. Now we can use optional or something like that.

Reply:
You can use exceptions instead of error codes.